aws-distro-opentelemetry-node-autoinstrumentation/test/patches/aws/services/bedrock.test.ts

```diff
@@ -0,0 +1,278 @@
+
+      await bedrock
+        .retrieve({
+          knowledgeBaseId: dummyKnowledgeBaseId,
+          retrievalQuery: { text: dummyQuery },
+        })
+        .catch((err: any) => {});
+
+      const testSpans: ReadableSpan[] = getTestSpans();
+      const describeSpans: ReadableSpan[] = testSpans.filter((s: ReadableSpan) => {
+        return s.name === 'BedrockAgentRuntime.Retrieve';
+      });
+      expect(describeSpans.length).toBe(1);
+      const creationSpan = describeSpans[0];
+      expect(creationSpan.attributes[AWS_ATTRIBUTE_KEYS.AWS_BEDROCK_AGENT_ID]).toBeUndefined();
+      expect(creationSpan.attributes[AWS_ATTRIBUTE_KEYS.AWS_BEDROCK_KNOWLEDGE_BASE_ID]).toBe(dummyKnowledgeBaseId);
+      expect(creationSpan.attributes[AWS_ATTRIBUTE_KEYS.AWS_BEDROCK_DATA_SOURCE_ID]).toBeUndefined();
+      expect(creationSpan.kind).toBe(SpanKind.CLIENT);
+    });
+  });
+
+  describe('InvokeAgent', () => {
+    it('adds agentId to span', async () => {
+      const dummyAgentId: string = 'ABCDEFGH';
+      const dummyAgentAliasId: string = 'HGFEDCBA';
+      const dummySessionId: string = 'ABC123AB';
+
+      nock(`https://bedrock-agent-runtime.${region}.amazonaws.com`).post('/').reply(200, {});
+
+      await bedrock
+        .invokeAgent({
+          agentId: dummyAgentId,
+          agentAliasId: dummyAgentAliasId,
+          sessionId: dummySessionId,
+        })
+        .catch((err: any) => {});
+
+      const testSpans: ReadableSpan[] = getTestSpans();
+      const describeSpans: ReadableSpan[] = testSpans.filter((s: ReadableSpan) => {
+        return s.name === 'BedrockAgentRuntime.InvokeAgent';
+      });
+      expect(describeSpans.length).toBe(1);
+      const creationSpan = describeSpans[0];
+      expect(creationSpan.attributes[AWS_ATTRIBUTE_KEYS.AWS_BEDROCK_AGENT_ID]).toBe(dummyAgentId);
+      expect(creationSpan.attributes[AWS_ATTRIBUTE_KEYS.AWS_BEDROCK_KNOWLEDGE_BASE_ID]).toBeUndefined();
+      expect(creationSpan.attributes[AWS_ATTRIBUTE_KEYS.AWS_BEDROCK_DATA_SOURCE_ID]).toBeUndefined();
+      expect(creationSpan.kind).toBe(SpanKind.CLIENT);
+    });
+  });
+});
+
+describe('Bedrock', () => {
+  let bedrock: Bedrock;
+  beforeEach(() => {
+    bedrock = new Bedrock({
+      region: region,
+      credentials: {
+        accessKeyId: 'abcde',
+        secretAccessKey: 'abcde',
+      },
+    });
+  });
+
+  describe('GetGuardrail', () => {
+    it('adds guardrailId to span', async () => {
+      const dummyGuardrailIdentifier: string = 'ABCDEFGH';
+
+      nock(`https://bedrock.${region}.amazonaws.com`).post('/').reply(200, {});
+
+      await bedrock
+        .getGuardrail({
+          guardrailIdentifier: dummyGuardrailIdentifier,
+        })
+        .catch((err: any) => {});
+
+      const testSpans: ReadableSpan[] = getTestSpans();
+      const describeSpans: ReadableSpan[] = testSpans.filter((s: ReadableSpan) => {
+        return s.name === 'Bedrock.GetGuardrail';
+      });
+      expect(describeSpans.length).toBe(1);
```

Comment:
nit (applies to all tests here)
- Instead of `describeSpans`, name could be `getGuardrailSpans`
- Instead of `creationSpan`, name could be `getGuardrailSpan`